test/arranger-conduit/Deposit.t.sol

```diff
@@ -34,48 +50,67 @@ contract ArrangerConduit_DepositFailureTests is ConduitAssetTestBase {
     }
 
     function test_deposit_insufficientFundsBoundary() public {
-        asset.mint(address(this), 99);
+        asset.mint(operator, 99);
+
+        vm.startPrank(operator);
+
         asset.approve(address(conduit), 100);
 
         vm.expectRevert(stdError.arithmeticError);
         conduit.deposit(ilk, address(asset), 100);
 
-        asset.mint(address(this), 1);
+        asset.mint(operator, 1);
 
         conduit.deposit(ilk, address(asset), 100);
     }
 
     function testFuzz_deposit_insufficientFundsBoundary(uint256 amount) public {
         vm.assume(amount != 0);
 
-        asset.mint(address(this), amount - 1);
+        asset.mint(operator, amount - 1);
+
+        vm.startPrank(operator);
+
         asset.approve(address(conduit), amount);
 
         vm.expectRevert(stdError.arithmeticError);
         conduit.deposit(ilk, address(asset), amount);
 
-        asset.mint(address(this), 1);
+        asset.mint(operator, 1);
 
         conduit.deposit(ilk, address(asset), amount);
     }
 
+    function test_deposit_transferFromRevert() public {
+        RevertingERC20 erc20 = new RevertingERC20("erc20", "ERC20", 18);
+
+        vm.etch(address(asset), address(erc20).code);
```

Comment:
Why does this need to be etched? You can submit any asset so why not just make the newly created erc20 the asset being passed in? This pattern of etching the code is not really that great in general and I think it's better to try and avoid.